harness/determined/deploy/gke/cli.py

```diff
@@ -0,0 +1,551 @@
+    if args.multi_node_pool:
+        if args.cpu_node_pool_name is None:
+            raise ValueError(
+                "--cpu-node-pool-name must be specified if using multiple node pools "
+                + "(--multi-node-pool flag is set).",
+            )
+    if args.gpu_coscheduler and args.preemption:
+        raise ValueError(
+            "--gpu-coscheduler and --preemptive-scheduler are mutually exclusive and cannot both be"
+            " specified",
+        )
+    if args.cpu_coscheduler and not args.gpu_coscheduler:
+        raise ValueError(
+            "To enable coscheduling on the CPU Node Pool, coscheduling must be enabled on the GPU"
+            " Node Pool. --cpu-coscheduler cannot be specified if --gpu-coscheduler isn't as well."
+        )
+    if args.cpu_coscheduler and not args.multi_node_pool:
+        raise ValueError(
+            "To enable coscheduling on the CPU Node Pool, multiple node pools must be used."
+            + " --cpu-coscheduler cannot be specified if --multi-node-pool isn't as well."
+        )
+
+
+def create_cluster(args: argparse.Namespace) -> None:
+    region = "-".join(args.zone.split("-")[:-1])
+    cmd = [
+        "gcloud",
+        "container",
+        "clusters",
+        "create",
+        args.cluster_name,
+        "--region",
+        region,
+        "--node-locations",
+        args.zone,
+        "--num-nodes=1",
+        "--machine-type={}".format(args.master_machine_type),
+    ]
+    subprocess.check_call(cmd, stdout=subprocess.DEVNULL)
+    create_nodepools(region, args)
+    if not args.no_make_bucket:
+        cmd = ["gsutil", "mb", "gs://{}".format(args.gcs_bucket)]
+        subprocess.check_call(cmd, stdout=subprocess.DEVNULL)
+
+
+def create_nodepools(region: str, args: argparse.Namespace) -> None:
+    cmd = [
+        "gcloud",
+        "container",
+        "node-pools",
+        "create",
+        args.gpu_node_pool_name,
+        "--cluster",
+        args.cluster_name,
+        "--accelerator",
+        "type={},count={}".format(args.gpu_type, args.gpus_per_node),
+        "--zone",
+        region,
+    ]
+    if args.gpu_coscheduler or args.preemption:
+        cmd += ["--num-nodes={}".format(args.max_gpu_nodes)]
+    else:
+        cmd += [
+            "--num-nodes=0",
+            "--enable-autoscaling",
+            "--min-nodes=0",
+            "--max-nodes={}".format(args.max_gpu_nodes),
+        ]
+    cmd += [
+        "--machine-type={}".format(args.agent_machine_type),
+        "--scopes=storage-full,cloud-platform",
+        "--node-labels=accelerator_type=gpu",
+    ]
+    if args.multi_node_pool:
+        cmd += ["--node-taints=gpuAvailable=True:NoSchedule"]
+    subprocess.check_call(cmd, stdout=subprocess.DEVNULL)
+    cmd = [
+        "kubectl",
+        "apply",
+        "-f",
+        defaults.K8S_NVIDIA_DAEMON,
+    ]
+    subprocess.check_call(cmd, stdout=subprocess.DEVNULL)
+    if args.multi_node_pool:
```

Comment:
nit: maybe suggest add some single blank lines between major "stages" in the long functions, e.g. a blank line between gpu and cpu sections here.

Reply:
I ended up breaking it into multiple methods, which helps with adding cpu only clusters!